internal/provider/burn_alert_resource_test.go

```diff
@@ -878,6 +921,43 @@ func burnAlertAccTestSetup(t *testing.T) (string, string) {
 	return dataset, slo.ID
 }
 
+func getNewDatasetAndSLO(t *testing.T) (string, string) {
+	t.Helper()
+
+	ctx := context.Background()
+	c := testAccClient(t)
+
+	dataset, err := c.Datasets.Create(ctx, &client.Dataset{
+		Name: acctest.RandomWithPrefix("test"),
+	})
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		c.Datasets.Delete(ctx, dataset.Slug)
```

Comment:
I believe this delete is failing as deletion protection isn't disabled after creation above